indexing-service/src/main/java/io/druid/indexing/overlord/TaskRunner.java

```diff
@@ -43,13 +45,18 @@
    *
    * @param taskid task ID to clean up resources for
    */
-  public void shutdown(String taskid);
+  void shutdown(String taskid);
 
-  public Collection<? extends TaskRunnerWorkItem> getRunningTasks();
+  Collection<? extends TaskRunnerWorkItem> getRunningTasks();
 
-  public Collection<? extends TaskRunnerWorkItem> getPendingTasks();
+  Collection<? extends TaskRunnerWorkItem> getPendingTasks();
 
-  public Collection<? extends TaskRunnerWorkItem> getKnownTasks();
+  Collection<? extends TaskRunnerWorkItem> getKnownTasks();
 
-  public Collection<ZkWorker> getWorkers();
+  /**
+   * Some runners are able to scale up and down their capacity in a dynamic manner. This returns stats on those activities
+   *
+   * @return ScalingStats if the runner has an underlying resource which can scale, Optional.absent() otherwise
+   */
+  Optional<ScalingStats> getScalingStats();
```

Comment:
similar to getWorkers() , getScalingStats() is also tied to some specific TaskRunners and ScalingStat looks like related much more to autoscaling . not sure if we are introducing same kind of coupling again via this.